src/java/main/org/apache/zookeeper/server/ZooKeeperServer.java

```diff
@@ -476,9 +474,12 @@ public ZooKeeperServerListener getZooKeeperServerListener() {
         return listener;
     }
 
+    // Visible for testing
+    static volatile int serverId = 1;
```

Comment:
This is really ugly. Why not using a quorum peer instead? Given you want to control the serverid.

Reply:
@phunt I'm not sure what you mean about quorum peer. I could find another way to set this value. Also, it's only for testing and clearly marked. What do you suggest?

Reply:
I looked at it a bit originally, iirc I figured that you'd pass it through in some way (during instance creation) rather than making it a global.

Reply:
As I recall that was extremely difficult. Config is not shared between different parts of the code. The only way to do that type of thing is via System defines. That's why I did this.

Reply:
It being a global seems to kick the problem down the road to the next person. I'm worried that setting this jvm wide could have an impact when developing tests in the future. That's why it caught my eye.

Reply:
Here's the issue @phunt - `serverId` is consumed by `ZooKeeperServer.createSessionTracker()`. For tests, the ZooKeeperServer is created by the **static** method `ClientBase.startServerInstance()`. So, _I'm_ the person down the road that has to deal with the static method created long ago. It would take significant re-write for `ClientBase.startServerInstance()` to be non-static and parameterized.

Reply:
That said - I could probably do it. Let me know. If you think it's important I'll see what I can do.